packages/funding-rate-proposer/src/proposer.js

```diff
@@ -278,12 +275,104 @@ class FundingRateProposer {
+    const publishableFundingRate = publishableProposals[0].proposedPrice.toString();
+
+    // Compare the proposed rates to the pricefeed's current rate.
+    const pricefeedPrice = await this._getPriceFeedAndPrice(fundingRateIdentifier);
+    if (!pricefeedPrice) return;
+    let shouldUpdateFundingRate = isDeviationOutsideErrorMargin(
+      this.toBN(publishableFundingRate), // ObservedValue
+      this.toBN(pricefeedPrice), // ExpectedValue
+      this.toBN(this.toWei("1")),
+      this.toBN(this.toWei(this.fundingRateErrorPercent.toString()))
+    );
+    if (shouldUpdateFundingRate) {
+      const apply = cachedContract.contract.methods.applyFundingRate();
+      this.logger.debug({
+        at: "PerpetualProposer#applyFundingRate",
+        message: "Applying new funding rate",
+        fundingRateIdentifier,
+        pricefeedPrice: pricefeedPrice,
+        pendingRateToPublish: publishableFundingRate,
+        allowedError: this.fundingRateErrorPercent
+      });
+      try {
+        // Get successful transaction receipt and return value or error.
+        const transactionResult = await runTransaction({
+          transaction: apply,
+          config: {
+            gasPrice: this.gasEstimator.getCurrentFastPrice(),
+            from: this.account
+            // Since this method is called within a Promise.all, it is sending transactions in parallel with other
+            // transactions, making it difficult to determine which nonce is the correct once to set for ynatm.
+            // Future work should build nonce management logic into the runTransactions method. See more here:
+            // https://github.com/ChainSafe/web3.js/issues/1846
+          }
+        });
+        let receipt = transactionResult.receipt;
+
+        const logResult = {
+          tx: receipt.transactionHash,
+          requester: contractAddress,
+          fundingRateIdentifier,
+          pricefeedPrice: pricefeedPrice,
+          pendingRateToPublish: publishableFundingRate
+        };
+        this.logger.info({
+          at: "PerpetualProposer#applyFundingRate",
+          message: "Applied new funding rate!🆒",
+          applyResult: logResult
+        });
+      } catch (error) {
+        const message = error.type === "call" ? "Cannot apply funding rate✋" : "Failed to apply funding rate🚨";
+        this.logger.error({
+          at: "PerpetualProposer#applyFundingRate",
+          message,
+          fundingRateIdentifier,
+          error
+        });
+        return;
+      }
+    } else {
+      this.logger.debug({
+        at: "PerpetualProposer#applyFundingRate",
+        message: "Skipping apply because pending rate is within allowed margin of error",
+        fundingRateIdentifier,
+        pricefeedPrice: pricefeedPrice,
+        pendingRateToPublish: publishableFundingRate,
+        allowedError: this.fundingRateErrorPercent
```

Comment:
does this need to be `.toString`ed? same above on line 325.

Reply:
No its a `Number`

Reply:
O I see. in the other places where this is used with `.toString()` it is being converted to a big number. all good in that case 👍